demos/multi_channel/yolo_v3/main.cpp

```diff
@@ -0,0 +1,584 @@
+    const int out_blob_h = static_cast<int>(blob->getTensorDesc().getDims()[2]);
+    const int out_blob_w = static_cast<int>(blob->getTensorDesc().getDims()[3]);
+    if (out_blob_h != out_blob_w)
+        throw std::runtime_error("Invalid size of output " + layer->name +
+        " It should be in NCHW layout and H should be equal to W. Current H = " + std::to_string(out_blob_h) +
+        ", current W = " + std::to_string(out_blob_h));
+    // --------------------------- Extracting layer parameters -------------------------------------
+    auto num = layer->GetParamAsInt("num");
+    auto coords = layer->GetParamAsInt("coords");
+    auto classes = layer->GetParamAsInt("classes");
+    std::vector<float> anchors;
+
+    try { anchors = layer->GetParamAsFloats("anchors"); } catch (...) {}
+    try {
+        auto mask = layer->GetParamAsInts("mask");
+        num = mask.size();
+
+        std::vector<float> maskedAnchors(num * 2);
+        for (int i = 0; i < num; ++i) {
+            maskedAnchors[i * 2] = anchors[mask[i] * 2];
+            maskedAnchors[i * 2 + 1] = anchors[mask[i] * 2 + 1];
+        }
+        anchors = maskedAnchors;
+    } catch (...) {}
+
+    auto side = out_blob_h;
+    auto side_square = side * side;
+    const float *output_blob = blob->buffer().as<InferenceEngine::PrecisionTrait<InferenceEngine::Precision::FP32>::value_type *>();
+    // --------------------------- Parsing YOLO Region output -------------------------------------
+    for (int i = 0; i < side_square; ++i) {
+        int row = i / side;
+        int col = i % side;
+        for (int n = 0; n < num; ++n) {
+            int obj_index = EntryIndex(side, coords, classes, n * side * side + i, coords);
+            int box_index = EntryIndex(side, coords, classes, n * side * side + i, 0);
+            float scale = output_blob[obj_index];
+            if (scale < threshold)
+                continue;
+            double x = (col + output_blob[box_index + 0 * side_square]) / side * resized_im_w;
+            double y = (row + output_blob[box_index + 1 * side_square]) / side * resized_im_h;
+            double height = std::exp(output_blob[box_index + 3 * side_square]) * anchors[2 * n + 1];
+            double width = std::exp(output_blob[box_index + 2 * side_square]) * anchors[2 * n];
+            for (int j = 0; j < classes; ++j) {
+                int class_index = EntryIndex(side, coords, classes, n * side_square + i, coords + 1 + j);
+                float prob = scale * output_blob[class_index];
+                if (prob < threshold)
+                    continue;
+                DetectionObject obj(x, y, height, width, j, prob,
+                        static_cast<float>(original_im_h) / static_cast<float>(resized_im_h),
+                        static_cast<float>(original_im_w) / static_cast<float>(resized_im_w));
+                objects.push_back(obj);
+            }
+        }
+    }
+}
+
+struct Object {
```

Comment:
I don't quite understand why there are two object structs. Wouldn't it be possible to remove one of them and use the other throughout the code?